File: revi_1/bubbley.py
# bubbble sort is a simple algo 
# compaers adjacent elemtns and swaps them if they are in the wronf order 
# gets name from bubble in water 
def bubble_sort(arr):
  
  n=len(arr)
  for i in range(n):
    swapped=False 
    for j in range(0,n-i-1):
      if arr[j] > arr[j+1]:
           arr[j],arr[j+1]=arr[j+1],arr[j]
          swapped=True 
      if not swapped:
        break
  return arr
  

File: revi_1/bubbley.go
package main

import "fmt"

func bubbleSort(arr []int) []int {
    n := len(arr)
    for i := 0; i < n; i++ {
        swapped := false
        for j := 0; j < n-i-1; j++ {
            if arr[j] > arr[j+1] {
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = true
            }
        }
        if !swapped {
            break
        }
    }
    return arr
}

func main() {
    example := []int{64, 34, 25, 12, 22, 11, 90}
    fmt.Println(bubbleSort(example))  
}
